src/main/java/com/github/manolo8/darkbot/gui/MapDrawer.java

```diff
@@ -79,10 +97,20 @@ public void setup(Main main) {
 
     protected void onPaint() {
         //background image is drawn first
-        if (main.config.BOT_SETTINGS.CUSTOM_BACKGROUND.ENABLED && main.config.BOT_SETTINGS.CUSTOM_BACKGROUND.IMAGE.getImage() != null) {
-            mapGraphics.g2.setComposite(AlphaComposite.getInstance(AlphaComposite.SRC_OVER, (float) main.config.BOT_SETTINGS.CUSTOM_BACKGROUND.OPACITY));
-            mapGraphics.g2.drawImage(main.config.BOT_SETTINGS.CUSTOM_BACKGROUND.IMAGE.getImage(), 0, 0, mapGraphics.width, mapGraphics.height, this);
-            mapGraphics.g2.setComposite(AlphaComposite.getInstance(AlphaComposite.SRC_OVER, 1f));
+        if (main.config.BOT_SETTINGS.CUSTOM_BACKGROUND.ENABLED) {
+            Image img = null;
+
+            if (main.config.BOT_SETTINGS.CUSTOM_BACKGROUND.USE_GAME_BACKGROUND)
+                img = backgroundImage;
```

Comment:
in here you can just check for `if hero.map.id != lastMapId` and update the image (or start a worker that downloads it for the future)